collector/mountstats_linux.go

```diff
@@ -507,37 +508,51 @@ func (c *mountStatsCollector) Update(ch chan<- prometheus.Metric) error {
 		return fmt.Errorf("failed to parse mountstats: %v", err)
 	}
 
+	mountsInfo, err := c.proc.MountInfo()
+	if err != nil {
+		return fmt.Errorf("failed to parse mountinfo: %v", err)
+	}
+
 	// store all seen nfsDeviceIdentifiers for deduplication
 	deviceList := make(map[nfsDeviceIdentifier]bool)
 
-	for _, m := range mounts {
+	for idx, m := range mounts {
 		// For the time being, only NFS statistics are available via this mechanism
 		stats, ok := m.Stats.(*procfs.MountStatsNFS)
+
 		if !ok {
 			continue
 		}
 
-		deviceIdentifier := nfsDeviceIdentifier{m.Device, stats.Transport.Protocol}
+		var mountAddress string
+		if idx < len(mountsInfo) {
+			// The mount entry order in the /proc/self/mountstats and /proc/self/mountinfo is the same.
```

Comment:
It seems like there is a small risk that a mount is added or removed between the calls to `MountStats` and `MountInfo` which could throw off the alignment between the two structs.  Would it be better to use a map and look up the Mountinfo by mountpoint?  Or is the risk minor enough that it's not necessary?

Reply:
The `/proc/self/mountstats` and `/proc/self/mountinfo` are supposed to be consistent, as they are both updated (and `/proc/self/mounts`) by the `mount` command. I don't think we would gain much by adding a defensive check here.